contrib/vspec2graphql/traverse_tree.py

```diff
@@ -0,0 +1,113 @@
+#
+# Copyright (C) 2022, Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+#
+# All files and artifacts in this repository are licensed under the
+# provisions of the license provided by the LICENSE file in this repository.
+#
+
+import re
+from anytree import RenderTree, PreOrderIter
+from vspec.model.vsstree import VSSNode, camel_back
+from vspec.model.constants import VSSDataType
+from anytree import PostOrderIter
+from typing import Dict
+
+from graphql import (
+    GraphQLSchema,
+    GraphQLObjectType,
+    GraphQLField,
+    GraphQLArgument,
+    GraphQLNonNull,
+    GraphQLString,
+    GraphQLList,
+    print_schema,
+    GraphQLSchema,
+    GraphQLObjectType,
+    GraphQLField,
+    GraphQLInt,
+    GraphQLFloat,
+    GraphQLBoolean,
+)
+
+GRAPHQL_TYPE_MAPPING = {
+    VSSDataType.INT8: GraphQLInt,
+    VSSDataType.INT8_ARRAY: GraphQLList(GraphQLInt),
+    VSSDataType.UINT8: GraphQLInt,
+    VSSDataType.UINT8_ARRAY: GraphQLList(GraphQLInt),
+    VSSDataType.INT16: GraphQLInt,
+    VSSDataType.INT16_ARRAY: GraphQLList(GraphQLInt),
+    VSSDataType.UINT16: GraphQLInt,
+    VSSDataType.UINT16_ARRAY: GraphQLList(GraphQLInt),
+    VSSDataType.INT32: GraphQLInt,
+    VSSDataType.INT32_ARRAY: GraphQLList(GraphQLInt),
+    VSSDataType.UINT32: GraphQLInt,
+    VSSDataType.UINT32_ARRAY: GraphQLList(GraphQLInt),
+    VSSDataType.INT64: GraphQLFloat,
+    VSSDataType.INT64_ARRAY: GraphQLList(GraphQLFloat),
+    VSSDataType.UINT64: GraphQLFloat,
+    VSSDataType.UINT64_ARRAY: GraphQLList(GraphQLFloat),
+    VSSDataType.FLOAT: GraphQLFloat,
+    VSSDataType.FLOAT_ARRAY: GraphQLList(GraphQLFloat),
+    VSSDataType.DOUBLE: GraphQLFloat,
+    VSSDataType.DOUBLE_ARRAY: GraphQLList(GraphQLFloat),
```

Comment:
Does GraphQL say anything about how many significant digits a float contain. I.e. is there a risk that we will loose some precision using GraphQLFloat for VSS Double?

Reply:
Double-precision, should be fine.